tools/hermes_adapter/providers/gemini_cli_client.py

```
@@ -0,0 +1,319 @@
+
+        stderr_lines: list[str] = []
+        stdout_lines: list[str] = []
+        open_streams = {"stdout", "stderr"}
+        started = time.monotonic()
+
+        try:
+            while open_streams:
+                remaining = timeout - (time.monotonic() - started)
+                if remaining <= 0:
+                    process.kill()
+                    raise TimeoutError(f"Gemini CLI timed out after {timeout}s")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)